### PR TITLE
feat: add consensus time to sampling

### PR DESCRIPTION
**Pull request by aloknerurkar**
### Checklist

- [ ] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [ ] I have filled out the description and linked the related issues.

### CODE REVIEW

**Review by notanatol — APPROVED** [review]

File: pkg/localstore/sampler.go [hunk 1]
```diff
@@ -104,6 +111,11 @@ func (db *DB) ReserveSample(ctx context.Context, anchor []byte, storageDepth uin
 					continue
 				}
 
+				if chItem.StoreTimestamp > consensusTime {
```
[the remark this answers is not in the document]
Comment (aloknerurkar) [reply pkg/localstore/sampler.go]: Its in Nanosecs. Thanks for pointing this out.

So now we will use the timestamp from the stamp which is also in Nanosecs. This is the same as what abel's PR will use to pass from the agent.

File: pkg/api/router.go [hunk 2]
```diff
@@ -312,12 +312,6 @@ func (s *Service) mountAPI() {
 		web.FinalHandlerFunc(s.healthHandler),
 	))
 
-	handle("/rchash/{depth}/{anchor}", web.ChainHandlers(
```
Comment (istae) [inline pkg/api/router.go:315]: can we keep this around for now, it's useful.

**Review by istae — CHANGES_REQUESTED** [review]

File: pkg/localstore/sampler.go [hunk 3]
```diff
@@ -104,6 +112,13 @@ func (db *DB) ReserveSample(ctx context.Context, anchor []byte, storageDepth uin
 					continue
 				}
 
+				// check if the timestamp on the postage stamp is not later than
+				// the consensus time.
+				if time.Duration(binary.BigEndian.Uint64(chItem.Timestamp))*time.Nanosecond > consensusTime {
```
Comment (metacertain) [inline pkg/localstore/sampler.go:117]: doesnt this lead to an extra 2-4 million conversions to time.Duration?

Comment (metacertain) [reply pkg/localstore/sampler.go:117]: also if time.duration is an alias for int64 isn't this conversion theoretically unsafe?

Comment (istae) [reply pkg/localstore/sampler.go:117]: Storing timestamps with int 64 bits is safe for decades to come, so no danger here.

**Review by metacertain — APPROVED** [review]

**Review by mrekucci — APPROVED** [review]

**Review by istae — APPROVED** [review]

**Pull request merged**